src/Analyzers/Core/Analyzers/NewLines/MultipleBlankLines/AbstractMultipleBlankLinesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.NewLines.MultipleBlankLines
+{
+    internal abstract class AbstractMultipleBlankLinesDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        protected AbstractMultipleBlankLinesDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.MultipleBlankLinesDiagnosticId,
+                   EnforceOnBuildValues.MultipleBlankLines,
+                   CodeStyleOptions2.DisallowMultipleBlankLines,
+                   LanguageNames.CSharp,
+                   new LocalizableResourceString(
+                       nameof(AnalyzersResources.Avoid_multiple_blank_lines), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)))
+        {
+        }
+
+        protected abstract bool IsEndOfLine(SyntaxTrivia trivia);
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxTreeAction(AnalyzeTree);
+
+        private void AnalyzeTree(SyntaxTreeAnalysisContext context)
+        {
+            var option = context.GetOption(CodeStyleOptions2.DisallowMultipleBlankLines, context.Tree.Options.Language);
+            if (!option.Value)
+                return;
```

Comment:
This option feels unnecessary. If the user wants to allow multiple blank lines, why not simply disable this rule?

Reply:
I don't understand your question. That's exactly what this check is for. If they disable the rule then this is what makes us bail out

Reply:
I meant that the user can do that `using dotnet_diagnostic.IDExxxx.severity = none`. So why an option that does the same thing needed?

Reply:
I guess i'm not 100% sure what checks still need to be in the analyzer.  It seems if the user explicitly sets the value to `true:...` we would need to still see that and bail out, no?  @mavasani for clarity.